Source/WebCore/mathml/MathMLScriptsElement.cpp

```diff
@@ -67,13 +68,19 @@ Ref<MathMLScriptsElement> MathMLScriptsElement::create(const QualifiedName& tagN
     return adoptRef(*new MathMLScriptsElement(tagName, document));
 }
 
+static const MathMLElement::Length defaultLength = { };
+
 const MathMLElement::Length& MathMLScriptsElement::subscriptShift()
 {
+    if (document().settings().coreMathMLEnabled())
```

Comment:
It seems these are only used here:
https://searchfox.org/wubkat/rev/ae89fb172f2f6dfa362aff8493e0e2978a9800d4/Source/WebCore/rendering/mathml/RenderMathMLScripts.cpp#281
https://searchfox.org/wubkat/rev/ae89fb172f2f6dfa362aff8493e0e2978a9800d4/Source/WebCore/rendering/mathml/RenderMathMLScripts.cpp#290

How about we skip these branches when MathML Core is enabled, and add debug assertions in MathMLScriptsElement::subscriptShift()/MathMLScriptsElement::superscriptShift() to make sure they are never called when the pref is enabled?

(this would avoid allocating this defaultLength constant)